### ISSUE TITLE
[Enhancement] Disable Dead Channels

### ISSUE THREAD

**Issue by nichcuta** [xteve-project/xTeVe#131]
Hi,

Currently the program detects dead links already and displays an "Availability" percentage for each playlist. I think that such data should also be visible in the log file meaning that each dead URL should be listed in the log.

Further to that, what if there was an option in the "Settings" menu to disable dead channels on the fly when the playlist is checked?  Obviously such check would also re-enable the channel if it is available again upon recheck.

I think the majority would prefer not seeing a channel rather than visible but not working.

I can create the above behaviour using Python and modifying the JSON file to disable the channels, but an integrated feature would make this awesome project even better.

Thanks,
Nic

**Comment by xteve-project** [comment]
The Availability column refers to the M3U / XMLTV file, not to the individual streams in the M3U.

Each M3U / XMLTV file is cached so that the latest version of the file is always available. If a server is currently down, the version from the cache is used. That is also in the log.

In order to recognize dead links, all streaming URLs from the M3U would have to be queried regularly, that would be several thousand requests a day. I can imagine that some providers would not like that.

Even if this function were implemented, it would not work with Plex. Channels that are deactivated in xTeVe are also deactivated in Plex. Plex does not automatically activate any channels if they are active again in xTeVe. Each channel would have to be activated manually in Plex.

**Comment by GroupMaster** [comment]
8 months ago I tried to clean things up with IPTV by tweaking things but at the end of the day when I thought I made progress, it just wasn't much better. I started playing around with connecting 2+ IPTV providers and several epg providers and I am close to having a noticeably better experience.  Guides are actually easy to deal with (just bulky), it's all about having more than one source of IPTV. This can be difficult to find because sometimes it's hard to tell who the source provider is. Many IPTV providers in one way or another use the same source and there are far fewer sources than providers, which include resellers. By keeping an up to date mapping of your provider's channel lists with your lineup, when a channel is called for they can be tested for basic availability. The first one that passes the test is used. A very large quantity of streams for each channel can be made available as long as one can be selected and used before plex times out. A small caveat of this, weather it is a simple php get_headers, an ffprobe command or a vlc connect will momentary require a +1 stream which can affect the playback of any currently active streams and make parallel testing of streams very difficult to manage. If the resources are available, a local streaming server can be set up to provide an offline video url to use in the event zero streams are available. This is where I am at in the testing phase. 

This can make the issue of down channel management a non-issue and channel customization easier. So the idea of testing the streams is good, it just needs to be applied correctly.
